File: 078.Subsets/clywin123/subsets.py
class Solution:
    # @param {integer[]} nums
    # @return {integer[][]}
    def subsets(self, nums):
        answer = []
        for n in range(int(math.pow(2,len(nums)))):
            rev = []
            rev = [(n>>i)&1 for i in range(32)]
            tmp = []
            i = 0
            for m in rev:
                if(m!=0):
                    tmp.append(nums[i])
                i+=1
            tmp = sorted(tmp)
            answer.append(tmp)
        return answer

File: 078.Subsets/Ddfish/Subsets.swift
class Solution {
	func subsets(_ nums: [Int]) -> [[Int]] {

		let nums = nums.sorted()
			let n = nums.count
			let subsetCount = 1 << n
			var result = [[Int]](repeatElement([Int](), count: subsetCount))

			for i in 0..<subsetCount {
				for j in 0..<n {
					if ( i & (1 << j)) != 0 {
						result[i].append(nums[j])
					}
				}
			}
		return result   
	}
}
